### PR TITLE
715 defence successfully configured message should not disappear after a delay

### PR DESCRIPTION
**Pull request by dhinrichs-scottlogic**
## Description
Previously, when a user changed or reset the text or number inputs of the defence and model configurations, a message would pop up for 3 seconds to indicate to the user that the change has been made. This can easily be missed, which is why I moved this message to the chat, where the user already gets updates when defences get toggled on/off. 
Additionally, I added an info message for the GPT model selection and the model sliders 

### Before:
image

### After:
image

## Notes
- changed two key/value pairs of the enum DEFENCE_ID: FILTER_USER_INPUT to INPUT_FILTERING and FILTER_BOT_OUTPUT to OUTPUT_FILTERING
- added the info messages to the chatHistory
- removed the old pop up info messages

## Concerns
- Due to the naming of the Information Defence, "defence" is repeated in the chat, there is already [a ticket](https://github.com/ScottLogic/prompt-injection/issues/776) to address this.
- There is currently no error messages for the user if the backend fails while we are changing a configuration. There's a [separate ticket ](https://github.com/ScottLogic/prompt-injection/issues/774)to address this. 
- The defence.name and the DEFENCE_ID are slightly different q/a LLM versus qa LLM. I couldn't change the enum to include the "/" in the enum key (which is what's being used in the function to update the config) and I didn't want to create a special conditional to change the output to the UI. Personally I think the difference between q/a and qa is acceptable for now. There is a general question about how useful the DEFENCE_ID enum is given that the defence names and the enum is almost identical, so this could become a bigger refactor. 

## Checklist

Have you done the following?

- [x] Linked the relevant Issue
- [ ] Added tests
- [x] Ensured the workflow steps are passing

### CODE REVIEW

**Comment by pmarsh-scottlogic** [comment]
this change interacts amusingly with this bug #742 
Not something to block this PR, but we should prioritise the bug once this is in

**Review by pmarsh-scottlogic — CHANGES_REQUESTED** [review]
I've not looked at the code yet, but this is a great change!

First thing I notice is that on refresh, or changing level then back again, these lovely new messages vanish! That's because when we add an info message to the messages state object, we must also update the chatHistory in the backend. See `addInfoMessage` defined in _MainComponent.tsx_.
Almost the same problem as #741 

I'll do a proper review next week

Edit: Turns out `addInfoMessage` in MainComponenet.tsx will add the message to the front and as well as the backend history. So the fix is to use `addInfoMessage` rather than `addChatMessage`.

File: frontend/src/components/MainComponent/MainComponent.tsx [hunk 1]
```diff
@@ -184,6 +184,9 @@ function MainComponent({
 			return defence;
 		});
 		setDefencesToShow(newDefences);
+		// add info message to chat
```
Comment (pmarsh-scottlogic) [inline frontend/src/components/MainComponent/MainComponent.tsx:187]: I'd prefer no comment here. I think the code reads well already without it!

File: frontend/src/components/MainComponent/MainComponent.tsx [hunk 2]
```diff
@@ -216,6 +219,9 @@ function MainComponent({
 				return defence;
 			});
 			setDefencesToShow(newDefences);
+			// add info message to chat
```
Comment (pmarsh-scottlogic) [inline frontend/src/components/MainComponent/MainComponent.tsx:222]: same here, don't think the comment is needed

**Review by pmarsh-scottlogic — CHANGES_REQUESTED** [review]
Super sleek!

Only some minor things from me

File: frontend/src/components/ModelBox/ModelConfigurationSlider.tsx [hunk 3]
```diff
@@ -2,22 +2,32 @@ import { Slider } from '@mui/material';
...
 	config: CustomChatModelConfiguration;
+	addChatMessage: (message: ChatMessage) => void;
 }) {
 	const [value, setValue] = useState<number>(config.value);
 	const [showInfo, setShowInfo] = useState<boolean>(false);
 
 	async function handleValueChange(_: Event, value: number | number[]) {
 		const val = Array.isArray(value) ? value[0] : value;
 		setValue(val);
+		addChatMessage({
```
Comment (pmarsh-scottlogic) [inline frontend/src/components/ModelBox/ModelConfigurationSlider.tsx:27]: You should import and use `addInfoMessage` instead. That'll sort out the refresh problem

File: frontend/src/components/ModelBox/ModelSelection.tsx [hunk 4]
```diff
@@ -22,6 +29,10 @@ function ModelSelection({ chatModelOptions }: { chatModelOptions: string[] }) {
 		if (!isSettingModel && selectedModel) {
 			const currentSelectedModel = selectedModel;
 			console.log(`selected model: ${currentSelectedModel}`);
+			addChatMessage({
+				type: CHAT_MESSAGE_TYPE.INFO,
+				message: `GPT model changed to: ${currentSelectedModel}`,
+			});
```
Comment (pmarsh-scottlogic) [inline frontend/src/components/ModelBox/ModelSelection.tsx:35]: You should import and use `addInfoMessage` instead. That'll sort out the refresh problem

File: frontend/src/components/MainComponent/MainComponent.tsx [hunk 5]
```diff
@@ -184,6 +184,9 @@ function MainComponent({
 			return defence;
 		});
 		setDefencesToShow(newDefences);
+		// add info message to chat
+		const displayedDefenceId = defenceId.replace(/_/g, ' ').toLowerCase();
+		addInfoMessage(`${displayedDefenceId} defence reset`);
```
Comment (chriswilty) [inline frontend/src/components/MainComponent/MainComponent.tsx:189]: You have these two lines duplicated now; may as well pull out into a separate function, which takes param 'reset' or 'configured' for the message to add.

**Review by chriswilty — CHANGES_REQUESTED** [review]
One primary concern, but all else good.

I assume there are no tests covering these components? If you want to learn some Testing Library style component tests, you might consider adding some for one of the simpler components, or some around the proposed validation changes. Doesn't need to be in this PR though.

**Comment by dhinrichs-scottlogic** [comment]
This all turned out much more complicated than anticipated, so after some discussion decided to approach it slightly differently and started again from scratch. 
New PR: https://github.com/ScottLogic/prompt-injection/pull/802

**Pull request closed without merging**